internal/image.go

```diff
@@ -224,3 +257,31 @@ func GetBuildpackageID(uri string) (string, error) {
 	}
 	return metadata.BuildpackageID, nil
 }
+
+func GetHighestPatch(patchVersion string, allVersions []string) (string, error) {
```

Comment:
It looks like this function is only used from within this package - can we unexport it?

Whether we keep it exported or not, we should backfill the remaining tests (e.g. the skipping over errors when upstream versions are not semver, and skipping over pre-release versions).

I recognize we cannot easily test the failure to create a `semver.NewVersion(patchVersion)` because any attempt to trigger that failure will likely cause `semver.NewConstraint` to fail earlier.

Reply:
Yea it is only used in this package - I can unexport for sure.